### PR TITLE
mgr/prometheus:Improve the pool metadata

### PR DESCRIPTION
**Pull request by pcuzner**
Adds percent_used and used_bytes to the per pool
metrics group, and add additional labels to the ceph_pool_metadata
metric
- compression_mode - so consumers can tell whether compression
  is active
- description - provide a text string showing the protection
  scheme of the pool (e.g. replica2, or ec:4+2
- type - text string showing replicated or erasure enabling
  easy filtering across the pools

These additional fields allow compression savings to be more
easily shown. The inclusion of percent_used ensures that any
prometheus based view of pool usage will ALWAYS match the CLI.

Fixes: https://tracker.ceph.com/issues/49049

Signed-off-by: Paul Cuzner <pcuzner@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by pcuzner** [comment]
With these changes in place the pools-overvire dashboard can look like this;

Screenshot-20210407165413-1906x928

File: src/pybind/mgr/prometheus/module.py [hunk 1]
```diff
@@ -820,9 +821,42 @@ def get_metadata_and_osd_status(self) -> None:
...
+                pool_type = "replicated"
+                description = f"replica:{pool['size']}"
+            elif pool['type'] == 3:
+                pool_type = "erasure"
+                name = pool.get('erasure_code_profile', '')
+                profile = ec_profiles.get(name, {})
+                if profile:
+                    description = f"ec:{profile['k']}+{profile['m']}"
+                else:
+                    description = "ec:unknown"
```
Comment (epuertat) [inline src/pybind/mgr/prometheus/module.py:840]: I like that it's consistent with the latest changes in the Pools page!

**Review by epuertat — APPROVED** [review]
Looks good, @pcuzner! 👍🏼

**Review by b-ranto — APPROVED** [review]

**Review by tchaikov — APPROVED** [review]

**Comment by tchaikov** [comment]
@pcuzner hi Paul, could you kindly add the Reviewed-by line(s) in the commit message of the merge commit when merging a PR in future?

**Comment by pcuzner** [comment]
Doh...apologies @tchaikov - will do!

**Pull request merged**